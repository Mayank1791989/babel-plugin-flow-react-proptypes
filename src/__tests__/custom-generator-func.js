const babel = require('babel-core');
const content = `
type FooProps = {
  bar: {
    a: string,
    b: number,
  }
};

const FooPropTypes = toPropTypes((t: FooProps) => t);
`;

it('generate props if generatorFunc enabled', () => {
  const res = babel.transform(content, {
    babelrc: false,
    presets: ['es2015', 'stage-1', 'react'],
    plugins: [
      'syntax-flow',
      [require('../'), { 'generatorFuncName': 'toPropTypes' }]
    ],
  }).code;
  expect(res).toMatchSnapshot();
});
