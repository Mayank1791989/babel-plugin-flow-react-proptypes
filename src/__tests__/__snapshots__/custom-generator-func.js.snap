// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate props if generatorFunc enabled 1`] = `
"\\"use strict\\";

var _propTypes = require(\\"prop-types\\");

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var FooPropTypes = {
  bar: _propTypes2.default.shape({
    a: _propTypes2.default.string.isRequired,
    b: _propTypes2.default.number.isRequired
  }).isRequired
};"
`;
